### DISCUSSION TITLE
Isolation.level property implementation not found for Consumer client

### DISCUSSION THREAD

**Discussion by rahul-hashedin** [zendesk/ruby-kafka#863] [General] [answered]
Hi,
Actually I have to support exactly one semantics for both producer and consumer end. I found it for the producer by using idempotent and transaction property but I am not able to find for the consumer.
As I researched, for supporting exactly-once semantics on the consumer side we have to set the isolation.level property to read_committed but I don't find any implementation for that. 
Can you suggest me some solution for this issue.
Thanks

**Comment by dasch** [comment, marked as answer]
This has not been implemented in ruby-kafka. Take a look at [rdkafka-ruby](https://github.com/appsignal/rdkafka-ruby) or a higher-level framework based on it, such as [Racecar](https://github.com/zendesk/racecar).

**Reply by acnovais** [reply]
Hi @dasch Isn't it the default supported mode [here](https://github.com/zendesk/ruby-kafka/blob/master/lib/kafka/protocol/fetch_request.rb#L53) or is there more requirements?
